File: Assignment/Abstraction/Expressions/NegateNode.cs
﻿namespace Assignment.Abstraction.Expressions
{
    internal struct NegateNode: INode
    {
        public NegateNode(INode innerNode)
        {
            this.InnerNode = innerNode;
        }

        public INode InnerNode { get; }

        public string DisplayName()
        {
            return nameof(NegateNode);
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/NegateNode.java
package ce305.abstraction.expressions;

import java.util.UUID;

import ce305.abstraction.INode;

public class NegateNode extends INode {
    public final INode innerNode;

    public NegateNode(INode innerNode) {
        super();
        this.innerNode = innerNode;
    }

    public NegateNode(INode innerNode, UUID id) {
        super(id);
        this.innerNode = innerNode;
    }

    @Override
    public String getDisplayName() {
        return NegateNode.class.getSimpleName();
    }
}